File: src/main/java/cc/duduhuo/git/conflict/Global.java
package cc.duduhuo.git.conflict;

import cc.duduhuo.git.conflict.model.ConflictItem;
import cc.duduhuo.git.conflict.model.MarkColor;
import com.intellij.openapi.editor.Document;
import com.intellij.openapi.editor.Editor;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/21 20:59
 * Description:
 * Remarks:
 * =======================================================
 */
public final class Global {
    // Is it highlighted
    public static Map<Editor, Boolean> sIsHighlightMap = new HashMap<>();
    // conflict item list
    public static Map<Document, List<ConflictItem>> sConflictItemMap = new HashMap<>();
    public static Map<Document, InDocumentListener> sDocumentListenerMap = new HashMap<>();
    // current color scheme
    public static MarkColor sCurrentColor = GlobalSettings.getMarkColor();
}

File: src/main/java/cc/duduhuo/git/conflict/Global.kt
package cc.duduhuo.git.conflict

import cc.duduhuo.git.conflict.model.ConflictItem
import cc.duduhuo.git.conflict.model.MarkColor
import cc.duduhuo.git.conflict.tool.delegate.CachedProperty
import com.intellij.openapi.editor.Document
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.editor.markup.RangeHighlighter

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/21 20:59
 * Description:
 * Remarks:
 * =======================================================
 */
object Global {
    // highlighter map to editor
    val highlighterMap: MutableMap<Editor, List<RangeHighlighter>> = mutableMapOf()

    // conflict item map to document
    val conflictItemMap: MutableMap<Document, List<ConflictItem>> = mutableMapOf()

    // document listener map to document
    val documentListenerMap: MutableMap<Document, InDocumentListener> = mutableMapOf()

    // current color scheme
    var currentColor: MarkColor by CachedProperty {
        val state = GlobalSettings.getPersistentState()
        state.markColors[state.schemeName]!!.toMarkColor()
    }

    // Whether automatically detects conflicts when a file is opened
    var autoDetectConflictsWhenFileOpened: Boolean by CachedProperty {
        val state = GlobalSettings.getPersistentState()
        state.autoDetectConflictsWhenFileOpened
    }
}
